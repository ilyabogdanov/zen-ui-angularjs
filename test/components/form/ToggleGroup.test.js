const div = document.createElement("div");
document.body.appendChild(div);

describe("directive: zen-toggle-group", function() {
    let element, scope;

    beforeEach(angular.mock.module("ZenUI"));
    afterEach(function () {
        while (div.firstChild) {
            div.removeChild(div.firstChild);
        }
    });

    it("should render", function () {
        inject(function($compile, $rootScope) {
            scope = $rootScope.$new();
            scope.model = null;
            element = ""+
                "<zen-toggle-group>" +
                    "<zen-toggle-radio ng-repeat=\"x in [1,2,3,4]\" zen-model=\"model\" zen-value=\"::x\">{{::x}}</zen-toggle-radio>" +
                "</zen-toggle-group>";
            element = $compile(element)(scope);
            scope.$digest();
            angular.element(div).append(element);
        })
        expect(div.childNodes).toHaveLength(1);
        expect(div.firstChild).toMatchSnapshot();
    });
    it("should render conjoined buttons", function () {
        inject(function($compile, $rootScope) {
            scope = $rootScope.$new();
            scope.model = null;
            element = ""+
                "<zen-toggle-group zen-conjoined>" +
                "<zen-toggle-radio ng-repeat=\"x in [1,2,3,4]\" zen-model=\"model\" zen-value=\"::x\">{{::x}}</zen-toggle-radio>" +
                "</zen-toggle-group>";
            element = $compile(element)(scope);
            scope.$digest();
            angular.element(div).append(element);
        })
        expect(div.childNodes).toHaveLength(1);
        expect(div.firstChild).toMatchSnapshot();
    });
});
