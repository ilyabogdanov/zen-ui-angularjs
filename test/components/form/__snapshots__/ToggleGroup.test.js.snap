// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directive: zen-toggle-group should render 1`] = `
<div
  class="ng-scope"
  ng-class="clazz"
  ng-transclude=""
>
  <!-- ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-3"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="1"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-3"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            1
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-5"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="2"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-5"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            2
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-7"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="3"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-7"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            3
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-9"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="4"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-9"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            4
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
</div>
`;

exports[`directive: zen-toggle-group should render conjoined buttons 1`] = `
<div
  class="ng-scope zen_ui__conjoined_buttons"
  ng-class="clazz"
  ng-transclude=""
  zen-conjoined=""
>
  <!-- ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-3"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="1"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-3"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            1
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-5"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="2"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-5"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            2
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-7"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="3"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-7"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            3
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    ng-repeat="x in [1,2,3,4]"
    zen-model="model"
    zen-value="::x"
  >
    <input
      class="zen_ui__toggle_button_input ng-pristine ng-untouched ng-valid ng-empty"
      id="zen-ui-id-9"
      name="zen-ui-id-1"
      ng-disabled="zenDisabled"
      ng-model="zenModel"
      ng-value="zenValue"
      type="radio"
      value="4"
    />
    <label
      class="zen_ui__button"
      data-palette="ordinary"
      for="zen-ui-id-9"
      ng-class="innerClazz"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          >
            4
          </span>
        </span>
      </span>
    </label>
    <div />
  </div>
  <!-- end ngRepeat: x in [1,2,3,4] -->
</div>
`;
